cmd/embedded-cluster/join.go

```diff
@@ -368,3 +419,74 @@ func runK0sInstallCommand(fullcmd string) error {
 	}
 	return nil
 }
+
+func waitForNode(ctx context.Context, kcli client.Client, hostname string) error {
+	loading := spinner.Start()
+	defer loading.Close()
+	loading.Infof("Waiting for node to join the cluster")
+	if err := kubeutils.WaitForNode(ctx, kcli, hostname); err != nil {
+		return fmt.Errorf("unable to wait for node: %w", err)
+	}
+	loading.Infof("Node has joined the cluster!")
+	return nil
+}
+
+func maybeEnableHA(ctx context.Context, kcli client.Client) error {
+	canEnableHA, err := canEnableHA(ctx, kcli)
+	if err != nil {
+		return fmt.Errorf("unable to check if HA can be enabled: %w", err)
+	}
+	if !canEnableHA {
+		return nil
+	}
+	logrus.Info("")
+	logrus.Info("When adding a third controller node, you have the option to enable high availability. This will migrate the data so that it is replicated across cluster nodes. Once enabled, you will be unable to have fewer than three controller nodes.")
+	logrus.Info("")
+	shouldEnableHA := prompts.New().Confirm("Do you want to enable high availability?", false)
+	if !shouldEnableHA {
+		return nil
+	}
+	logrus.Info("")
+	return enableHA(ctx, kcli)
+}
+
+// canEnableHA checks if high availability can be enabled in the cluster.
+func canEnableHA(ctx context.Context, kcli client.Client) (bool, error) {
+	installation, err := kubeutils.GetLatestInstallation(ctx, kcli)
+	if err != nil {
+		return false, fmt.Errorf("unable to get latest installation: %w", err)
+	}
+	if installation.Spec.HighAvailability {
+		return false, nil
+	}
+	var nodes corev1.NodeList
+	opts := &client.ListOptions{LabelSelector: labels.Set(config.ControllerLabels()).AsSelector()}
```

Comment:
What if we add more labels to `ControllerLabels` later? Wouldn't that affect existing nodes that don't have the additional labels? How about using the `node-role.kubernetes.io/control-plane` label instead? That's what we're using for node affinity rules for rqlite and seaweedfs.

Reply:
updated the selector in https://github.com/replicatedhq/embedded-cluster/pull/657/commits/a6b28b202b1d01742a034c00aebc953238705ca9